### PR TITLE
Adding Markdown syntax highlighting

### PR DESCRIPTION
**Pull request by jlahoda**
Just a simple plugin that provides syntax highlighting for Markdown using a Textmate grammar (from VS Code).

Matthias, could you please check if this looks OK w.r.t. licenses?

Thanks!

### CODE REVIEW

File: nbbuild/licenses/CC0-markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
+DEDICATED TO THE PUBLIC DOMAIN BY DUSTIN CURTIS
+
+To the extent possible under law, Dustin Curtis has waived all copyright and related or neighboring rights to Markdown Mark.
+
+The Markdown Mark has been dedicated to the public domain. It is protected by the Creative Commons CC0 Universal Public Domain Dedication license. You can read the entire license below or at https://creativecommons.org/publicdomain/zero/1.0/deed.en.
+
+CC0 UNIVERSAL PUBLIC DOMAIN DEDICATION LICENSE
```
Comment (matthiasblaesing) [inline nbbuild/licenses/CC0-markdown]: Isn't this identical to `CC0-v10`? From my understanding of CC0, the author releases all copyrights/restricts he/she could pose upon a work. For the npm the icon is a recolored version of a CC0 basis - that derived file was marked as ALv2 and the header gives the history.

**Comment by gilbertoca** [comment]
Will this plug-in conflict with this one https://github.com/madflow/flow-netbeans-markdown ?

File: nbbuild/licenses/CC0-markdown (hunk 1 above)
Comment (jlahoda) [reply nbbuild/licenses/CC0-markdown]: I think we can use CC0-v10 - I did that in the latest update (there are some formatting changes, and the initial caption is different).

Thanks for the comment!

**Comment by jlahoda** [comment]
> Will this plug-in conflict with this one https://github.com/madflow/flow-netbeans-markdown ?

Uh, I didn't know about this plugin - I change the code here so that it can (/is) suppressed when that module is installed.

I would also be awesome if that module could be contributed here!

Thanks for the comment!

**Comment by neilcsmith-net** [comment]
I've retriggered failing Travis tests. What's the plan with this one?  Merge for 11.2 or wait?

**Comment by jlahoda** [comment]
I was inclined to incorporate the external plugin if possible, but didn't actually get to really look into that. So I'll leave it up to you if we want something simple for 11.2, or if it can wait for the next release. (I can wait, FWIW.)

**Comment by Chris2011** [comment]
@jlahoda is this already implemented into 11.3? I opened a MD file and no syntax highlighting is there.

**Comment by Chris2011** [comment]
Ahh sry, I see, it was not merged. My fault :)

**Comment by MiguelVis** [comment]
Hi! Any news on this? I really want Markdown support in NetBeans! Thanks.

**Comment by geertjanw** [comment]
Shall we include this in 12.4?

**Comment by geertjanw** [comment]
If this conflict is fixed, we can merge it, @jlahoda, or should we close it?

**Comment by jkost** [comment]
Another, not so popular, [plugin](https://github.com/jkost/MarkdownViewerForNetBeans) based on [flexmark](https://github.com/vsch/flexmark-java). 
Jan, since you have always been the hidden source behind NetBeans development, maybe you could shed some light on how we could achieve something like the below, i.e. a split window that displays both the source and the preview. MultiView window doesn't provide this functionality unfortunately, and it is not obvious to add it to it, or the solution is to simply not use MultiView maybe?

Fig2

**Pull request merged**